assets/js/vue-apps/components/budget-input.vue

```diff
@@ -10,16 +10,9 @@ export default {
         maxItems: { type: Number, required: true },
         budgetData: { type: Array, required: false }
     },
-    mounted() {
-        if (this.budgetData) {
-            this.budgetRows = this.budgetData;
-        } else {
-            this.addNewItem();
-        }
-    },
     data() {
         return {
-            budgetRows: [],
+            budgetRows: this.budgetData || [{ item: '', cost: '' }],
```

Comment:
I know this is repeating the object schema, but in testing using mounted and updating state caused issues, setting explicitly seems to be the way to go, even at the expense of a bit of duplication.

Reply:
👍